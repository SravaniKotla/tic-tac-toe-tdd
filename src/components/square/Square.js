import React from 'react';
import PropTypes from 'prop-types';

const Square = (props) => {
    return (
      <button className="square" onClick={props.onClick}>
        
      </button>
    );
  }
  Square.propTypes = { 
    onClick: PropTypes.func,
    value: PropTypes.string
  };
  export default Square;