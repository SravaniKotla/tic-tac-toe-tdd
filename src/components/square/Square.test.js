import React from 'react';
import { shallow } from 'enzyme';
import Square from './Square';

describe('Square Component', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Square />));

  //snapshot testing
  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
  
  it('should render a button', () => {
    expect(wrapper.find('button.square').length).toEqual(1);
  });

});