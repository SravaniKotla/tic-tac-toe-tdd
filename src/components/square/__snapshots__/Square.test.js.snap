// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Component should render correctly 1`] = `ShallowWrapper {}`;
